readthedocs/organizations/forms.py

```diff
@@ -0,0 +1,302 @@
+from readthedocs.organizations.constants import ADMIN_ACCESS, READ_ONLY_ACCESS
+from readthedocs.organizations.models import (
+    Organization,
+    OrganizationOwner,
+    Team,
+    TeamInvite,
+    TeamMember,
+)
+
+
+class OrganizationForm(forms.ModelForm):
+
+    """
+    Base organization form.
+
+    :param user: User instance, responsible for ownership of Organization
+    :type user: django.contrib.auth.models.User
+    """
+
+    class Meta:
+        model = Organization
+        fields = ['name', 'email', 'description', 'url']
+        labels = {
+            'name': _('Organization Name'),
+            'email': _('Billing Email'),
+        }
+
+    # Don't use a URLField as a widget, the validation is too strict on FF
+    url = forms.URLField(
+        widget=forms.TextInput(attrs={'placeholder': 'http://'}),
+        label=_('Site URL'),
+        required=False,
+    )
+
+    def __init__(self, *args, **kwargs):
+        try:
+            self.user = kwargs.pop('user')
+        except KeyError:
+            raise TypeError(
+                'OrganizationForm expects a `user` keyword argument',
+            )
+        super().__init__(*args, **kwargs)
+
+    def clean_name(self):
+        """Raise exception on duplicate organization."""
+        name = self.cleaned_data['name']
+        if self.instance and self.instance.name and name == self.instance.name:
+            return name
+        if Organization.objects.filter(slug=slugify(name)).exists():
+            raise forms.ValidationError(
+                _('Organization %(name)s already exists'),
+                params={'name': name},
+            )
+        return name
+
+
+class OrganizationSignupFormBase(OrganizationForm):
+
+    """
+    Simple organization creation form.
+
+    This trims down the number of inputs required to create a new organization.
+    This is used on the initial organization signup, to keep signup terse.
+
+    :param user: User instance, responsible for ownership of Organization
+    :type user: django.contrib.auth.models.User
+    """
+
+    class Meta:
+        model = Organization
+        fields = ['name', 'email']
+        labels = {
+            'name': _('Organization Name'),
+            'email': _('Billing Email'),
+        }
+
+    url = None
+
+    @staticmethod
+    def _create_default_teams(organization):
+        organization.teams.create(name='Admins', access=ADMIN_ACCESS)
+        organization.teams.create(name='Read Only', access=READ_ONLY_ACCESS)
+
+    def save(self, commit=True):
+        org = super().save(commit)
+
+        # If not commiting, we can't save M2M fields
+        if not commit:
+            return org
+
+        # Add default teams
+        OrganizationOwner.objects.create(
+            owner=self.user,
+            organization=org,
+        )
+        self._create_default_teams(org)
+        return org
+
+
+class OrganizationSignupForm(SettingsOverrideObject):
```

Comment:
Do we already know that we need to modify the behavior from Corporate? Is there a PR that is already making usage of this?

Reply:
there is already a PR open for that